### PR TITLE
RPC and view modification to keep relation between proxy and implementation contract during verification

### PR DESCRIPTION
**Pull request by aaitor**
## Motivation

This issue implements:
https://github.com/celo-org/celo-monorepo/issues/1186

### Enhancements
This issue allows to:
* Specify during the verification process if a contract is behind a proxy address (RPC api & verification forms)
* Amend the visualization when a contract is a implementation or proxy 
* Table index to keep that tracking with the proper table index and migration script

### CODE REVIEW

File: apps/block_scout_web/lib/block_scout_web/controllers/api/rpc/contract_controller.ex [hunk 1]
```diff
@@ -9,13 +9,15 @@ defmodule BlockScoutWeb.API.RPC.ContractController do
...
   def verify(conn, %{"addressHash" => address_hash} = params) do
+
     with {:params, {:ok, fetched_params}} <- {:params, fetch_verify_params(params)},
          {:format, {:ok, casted_address_hash}} <- to_address_hash(address_hash),
          {:params, external_libraries} <-
            {:params, fetch_external_libraries(params)},
          {:publish, {:ok, _}} <-
            {:publish, Publisher.publish(address_hash, fetched_params, external_libraries)} do
       address = Chain.address_hash_to_address_with_source_code(casted_address_hash)
+      Logger.debug(Map.get(fetched_params, "proxy_address"))
```
Comment (nambrot) [inline apps/block_scout_web/lib/block_scout_web/controllers/api/rpc/contract_controller.ex]: Should this line be here? Or at least expand in information?

File: apps/explorer/lib/explorer/smart_contract/publisher.ex [hunk 2]
```diff
@@ -26,6 +27,8 @@ defmodule Explorer.SmartContract.Publisher do
   def publish(address_hash, params, external_libraries \\ %{}) do
     params_with_external_libaries = add_external_libraries(params, external_libraries)
 
+    Logger.debug(Map.get(params_with_external_libaries, "proxy_address"))
```
Comment (nambrot) [inline apps/explorer/lib/explorer/smart_contract/publisher.ex]: Same

File: apps/explorer/priv/repo/migrations/20191022154112_proxy_contract_index.exs [hunk 3]
```diff
@@ -0,0 +1,9 @@
+defmodule Explorer.Repo.Migrations.CreateProxyContractIndex do
+  use Ecto.Migration
+
+  def change do
+    drop_if_exists index("proxy_contract", [:id])
+	  create(unique_index(:proxy_contract, [:proxy_address, :implementation_address]))
```
Comment (nambrot) [inline apps/explorer/priv/repo/migrations/20191022154112_proxy_contract_index.exs]: indention?

**Comment by nambrot** [comment]
What happens if the implementation changes?

**Comment by aaitor** [comment]
> What happens if the implementation changes?

```
    %ProxyContract{}
    |> ProxyContract.changeset(params)
    |> repo.insert(
         on_conflict: :replace_all,
         conflict_target: [:proxy_address, :implementation_address])
```

this must handle the upsert, but I will make some additional tests to validate it

**Comment by aaitor** [comment]
> What happens if the implementation changes?

Modify the table `index` to handle properly the upserts

**Pull request merged**